handlers/main.yml

```diff
@@ -1,12 +1,11 @@
 ---
 - name: Restart
-  shell: reboot
+  command: /sbin/shutdown -r +0 'Ansible Reboot triggered'
   async: 0
   poll: 0
   ignore_errors: true
-  sudo: yes
+  become: true
 
 - name: Wait for restart to finish
-  local_action: wait_for host={{ inventory_hostname }}
-                state=started
-  sudo: no
+  local_action: wait_for host={{ ansible_ssh_host }} state=started port=22 delay=50 timeout=120
```

Comment:
Is there a reason why you explicitly set the port here? If so, would it be better to use the `ansible_ssh_port` variable perhaps?